base/operators.jl

```diff
@@ -1149,40 +1149,68 @@ julia> filter(!isletter, str)
 !(f::ComposedFunction{typeof(!)}) = f.inner #allows !!f === f
 
 """
-    Fix1(f, x)
+    Fix{n}(f, x)
+    Fix{:kw}(f, x)
+    Fix(f; kw=x)
 
-A type representing a partially-applied version of the two-argument function
-`f`, with the first argument fixed to the value "x". In other words,
-`Fix1(f, x)` behaves similarly to `y->f(x, y)`.
+A type representing a partially-applied version of a function `f`, with the argument
+"x" fixed at argument `n::Int` or keyword `kw::Symbol`.
+In other words, `Fix{3}(f, x)` behaves similarly to
+`(y1, y2, y3) -> f(y1, y2, x, y3)` for the 4-argument function `f`.
 
-See also [`Fix2`](@ref Base.Fix2).
+You may also use this to fix keyword arguments. For example, `Fix(g; a=2)` behaves
+similarly to `x -> g(x; a=2)` for a function `g` with one argument and one keyword argument.
+You can also write this as `Fix{:a}(g, 2)`.
+
+!!! note
+    When nesting multiple `Fix`, note that the `N` in `Fix{N}` is _relative_ to the current
+    available arguments, rather than an absolute ordering on the target function. For example,
+    `Fix{1}(Fix{2}(f, 4), 4)` fixes the first and second arg, while `Fix{2}(Fix{1}(f, 4), 4)`
+    fixes the first and third arg.
 """
-struct Fix1{F,T} <: Function
+struct Fix{N,F,T} <: Function
     f::F
     x::T
 
-    Fix1(f::F, x) where {F} = new{F,_stable_typeof(x)}(f, x)
-    Fix1(f::Type{F}, x) where {F} = new{Type{F},_stable_typeof(x)}(f, x)
+    function Fix{N}(f::F, x) where {N,F}
+        if N isa Int && N < 1
+            throw(ArgumentError("expected `N` in `Fix{N}` to be integer greater than 0"))
+        elseif !(N isa Union{Int,Symbol})
+            throw(ArgumentError("expected type parameter in `Fix` to be `Int` or `Symbol`, but got type=$(typeof(N))"))
+        end
+        new{N,_stable_typeof(f),_stable_typeof(x)}(f, x)
+    end
+end
+function Fix(f::F; kws...) where {F}
+    length(kws) != 1 && throw(ArgumentError("`Fix` expects exactly one argument or keyword argument"))
+    Fix{only(keys(kws))}(f, only(values(kws)))
+end
+
+function (f::Fix{N})(args::Vararg{Any,M}; kws...) where {N,M}
+    if N isa Symbol
+        N in keys(kws) && throw(ArgumentError("found duplicate keyword argument passed to `Fix`"))
```

Comment:
I think it should error because `Fix` is supposed to _fix_ an argument or keyword argument, rather than act as a default that can be overridden. I think to do otherwise would be surprising behavior and likely result in some hard-to-find bugs, whereas throwing an error immediately gives feedback to the user so they can work around the imposed limitation (if they need to), or notice a bug from duplicate kwargs.